frontend/src/App.tsx

```diff
@@ -438,6 +438,7 @@ export class App extends PureComponent<Props, State> {
 
     document.title = `${reportName} · Streamlit`
 
+    MetricsManager.current.setReportHash(newReportHash)
     MetricsManager.current.clearDeltaCounter()
 
     MetricsManager.current.enqueue("updateReport", {
```

Comment:
`updateReport` passes `newReportHash` as data which wouldn't be necessary anymore with these changes

Reply:
Correct.

Reply:
Your PR should make that change

Reply:
I'd like a confirmation from @cobradford or @tvst before I remove it.

Reply:
Done!